components/brave_wallet/browser/sns_resolver_task.cc

```diff
@@ -28,15 +25,73 @@
 #include "brave/components/brave_wallet/browser/solana_requests.h"
 #include "brave/components/brave_wallet/browser/solana_response_parser.h"
 #include "brave/components/brave_wallet/common/brave_wallet_constants.h"
+#include "brave/components/brave_wallet/common/eth_address.h"
+#include "brave/components/brave_wallet/common/solana_address.h"
 #include "build/build_config.h"
 #include "crypto/sha2.h"
 #include "third_party/abseil-cpp/absl/cleanup/cleanup.h"
 #include "third_party/boringssl/src/include/openssl/curve25519.h"
 
 namespace brave_wallet {
 
+struct SnsFetchRecordQueueItem {
+  std::string record;
+  SnsRecordsVersion version;
+  SolanaAddress record_address;
+};
+
 namespace {
 
+std::vector<SnsFetchRecordQueueItem> GetWalletAddressQueueRecords() {
+  std::vector<SnsFetchRecordQueueItem> result;
+  result.reserve(2);
+
+  result.emplace_back();
+  result.back().record = kSnsSolRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV2;
+
+  result.emplace_back();
+  result.back().record = kSnsSolRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV1;
+
+  return result;
+}
+
+std::vector<SnsFetchRecordQueueItem> GetUrlQueueRecords() {
+  std::vector<SnsFetchRecordQueueItem> result;
+  result.reserve(4);
+
+  result.emplace_back();
+  result.back().record = kSnsUrlRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV2;
+
+  result.emplace_back();
+  result.back().record = kSnsIpfsRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV2;
+
+  result.emplace_back();
+  result.back().record = kSnsUrlRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV1;
+
+  result.emplace_back();
+  result.back().record = kSnsIpfsRecord;
+  result.back().version = SnsRecordsVersion::kRecordsV1;
+
+  return result;
+}
+
+SolanaAddress GetEmptyNameClass() {
+  return SolanaAddress::ZeroAddress();
+}
+
+SolanaAddress GetCentalStateSnsRecordsNameClass() {
+  // https://github.com/Bonfida/sns-sdk/blob/e930b83/rust-crates/sns-sdk/src/record/mod.rs#L10
+  auto result =
+      SolanaAddress::FromBase58("2pMnqHvei2N5oDcVGCRdZx48gqti199wr5CsyTTafsbo");
+  CHECK(result);
```

Comment:
nit: for my context, what's the value in checking this? It seems like it would always be correct since it's a constant value.